### ISSUE TITLE
paho cleansession set false,reconnect cant receive subscription

### ISSUE THREAD

**Issue by fangsj** [emqx/emqx#2512]
#### Environment

- OS: centos7
- Erlang/OTP: R21/10.3.2
- EMQ: 3.1.0

#### Description

* Using PAHO mqtt client, set CleanSession = false*

1. Connect emqx and do not restart emqx.

2. Restart emqx, mqttclient reconnect emqx but can not subscribe, no subscription will be received after startup.

3. Restart emqx restart, mqttclient reconnect emqx but can not subscribe, no subscription will be received after startup

4. Shut down emqx, close mqttclient, restart emqx, restart mqttclient, can subscribe successfully

**Comment by Gilbert-Wong** [comment]
Hi, @fangsj , which paho library do you use and how do you write your client code ? emqx 3.0.1 has passed paho tests with paho python library. Could you please give us more details ?

**Comment by fangsj** [comment]
use ```org.eclipse.paho:org.eclipse.paho.client.mqttv3:1.2.1```
```java
client = new AdvancedMqttClient(mqttConfiguration.getHost(), mqttConfiguration.getClientId(), new MemoryPersistence());
            MqttConnectOptions options = new MqttConnectOptions();
            options.setCleanSession(mqttConfiguration.isCleanSession());
            options.setUserName(mqttConfiguration.getUsername());
            options.setPassword(mqttConfiguration.getPassword().toCharArray());
            options.setConnectionTimeout(mqttConfiguration.getTimeout());
            options.setKeepAliveInterval(mqttConfiguration.getKeepalive());
            options.setMaxInflight(200);
            options.setMqttVersion(MQTT_VERSION_3_1_1);
            options.setAutomaticReconnect(true);
            // 2分钟
            options.setMaxReconnectDelay(120000);
            try {
                mqttCallback.setMqttPushClient(this);
                client.setCallback(mqttCallback);
                client.connect(options);
            } catch (MqttException e) {
                log.error(e.getMessage(), e);
            }
```
##### restart emqx
![image](https://user-images.githubusercontent.com/8734881/57211590-5d172c00-7013-11e9-90aa-9de711551212.png)

##### paho client info，cant receive subscription
![image](https://user-images.githubusercontent.com/8734881/57211562-440e7b00-7013-11e9-8160-5605e2979a6b.png)

**Comment by Gilbert-Wong** [comment]
First, from your code, I can not conclude that if it is your client problem triggering your problem.

Second, from your description, it seems that your client reconnected to emqx after emqx restarted. In opensource edition of emqx, if you restart emqx, even though your client set cleansession as false, you still cannot receive the messages from your subscription topic your client subscribed before emqx restarted.

Third, your client cannot subscribe to emqx. It makes me confused. Could you please  rotate emqx log level to debug, reproduce this issue and send me your log file to <gilbertw@emqx.io> to help us resolve it ?

**Comment by fangsj** [comment]
I found that if I executed "emqx restart" and did not restart my PAHO client, I could not subscribe to it again, but if I executed "emqx stop", then "emqx start" and did not restart the client, I could continue to subscribe to it again.

**Comment by MrPec** [comment]
I think I have a similar issue with Node-Red (1.0.1) and emqx 4.0.7 (also with 3.0.1 we used before).

1. Subscribe to several topics with Node-Red, clean session set to false
2. Restart emqx
3. Node-Red reconnects  after a short period and shows "connected"
4. We don't receive any messages

We have to also restart the client in order to subscrive succesfully. We replaced emqx with vernemq and we don't run into that issue.

Any ideas?
